manipmongo/utils_test.go

```diff
@@ -246,27 +251,44 @@ func Test_makeFieldsSelector(t *testing.T) {
 			"empty",
 			args{
 				[]string{},
+				nil,
 			},
 			nil,
 		},
 		{
 			"nil",
 			args{
 				nil,
+				nil,
 			},
 			nil,
 		},
 		{
 			"only empty",
 			args{
 				[]string{"", ""},
+				nil,
 			},
 			nil,
 		},
+		{
+			"translate fields from provided spec",
+			args{
+				fields: []string{"FieldA"},
+				spec: map[string]elemental.AttributeSpecification{
+					"FieldA": {
```

Comment:
- what happens when FieldA is not in the map?
- what happens when FieldA.BSONField is empty?

Reply:
Good questions!

If the attribute is not in the map and a spec has been configured, it will default to using the value of the attribute provided (_just like the existing default behavior_).

I added new unit tests to cover this exact scenario ✨